### PR TITLE
Precommit batch balancer support/config

### PR DESCRIPTION
**Pull request by magik6k**
Implements #7391

### CODE REVIEW

File: extern/storage-sealing/precommit_batch.go [hunk 1]
```diff
@@ -210,6 +234,82 @@ func (b *PreCommitBatcher) maybeStartBatch(notif bool) ([]sealiface.PreCommitBat
...
+		if (*avail).LessThan(big.Zero()) {
+			*avail = big.Zero()
+		}
+	}
+
+	goodFunds := big.Add(deposit, big.Int(b.feeCfg.MaxPreCommitGasFee))
+
+	from, _, err := b.addrSel(b.mctx, mi, api.PreCommitAddr, goodFunds, deposit)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("no good address to send commit message from: %w", err)
```
Comment (arajasek) [inline extern/storage-sealing/precommit_batch.go:302]:
```suggestion
		return cid.Undef, xerrors.Errorf("no good address to send precommit message from: %w", err)
```

**Comment by arajasek** [comment]
Almost good to go, needs some work once https://github.com/filecoin-project/specs-actors/pull/1497 lands

File: extern/storage-sealing/precommit_batch_test.go [hunk 2]
```diff
@@ -54,10 +54,11 @@ func TestPrecommitBatcher(t *testing.T) {
 			WaitDealsDelay:            time.Hour * 6,
 			AlwaysKeepUnsealedCopy:    true,
 
-			BatchPreCommits:     true,
-			MaxPreCommitBatch:   maxBatch,
-			PreCommitBatchWait:  24 * time.Hour,
-			PreCommitBatchSlack: 3 * time.Hour,
+			BatchPreCommits:            true,
+			MaxPreCommitBatch:          maxBatch,
+			PreCommitBatchWait:         24 * time.Hour,
```
Comment (jennijuju) [inline extern/storage-sealing/precommit_batch_test.go:59]: should we take this chance to lower it? SPs have been complaining this is too long

**Comment by jennijuju** [comment]
should we take this chance to lower `PreCommitBatchWait`? SPs have been complaining this is too long

**Comment by arajasek** [comment]
@jennijuju Sure, but I dunno what we want...Wanna push a commit to this branch / gimme a new value?

**Comment by jennijuju** [comment]
> @jennijuju Sure, but I dunno what we want...Wanna push a commit to this branch / gimme a new value?

we can ignore yesterday-me

**Review by arajasek — APPROVED** [review]

**Pull request merged**